musl does not define an identifier macro like __GLIBC__ or __UCLIBC
which is right. So we make sure that we do the right thing and not assume
defines

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: pciutils-3.2.1/lib/pread.h
===================================================================
--- pciutils-3.2.1.orig/lib/pread.h	2014-04-27 03:00:31.918719246 -0700
+++ pciutils-3.2.1/lib/pread.h	2014-04-27 03:24:14.972059749 -0700
@@ -12,7 +12,8 @@
  *  don't define it.
  */
 
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0
+#if defined(__GLIBC__)
+#if __GLIBC__ == 2 && __GLIBC_MINOR__ > 0
 /* glibc 2.1 or newer -> pread/pwrite supported automatically */
 
 #elif defined(i386) && defined(__GLIBC__)
@@ -59,6 +60,7 @@
 make_rw_glue(write)
 #define PCI_HAVE_DO_READ
 #endif
+#endif
 
 #ifndef PCI_HAVE_DO_READ
 #define do_read(d,f,b,l,p) pread(f,b,l,p)
